x/ibc/02-client/keeper/keeper.go

```diff
@@ -177,20 +177,29 @@ func (k Keeper) GetLatestClientConsensusState(ctx sdk.Context, clientID string)
 
 // GetClientConsensusStateLTE will get the latest ConsensusState of a particular client at the latest height
 // less than or equal to the given height
-func (k Keeper) GetClientConsensusStateLTE(ctx sdk.Context, clientID string, maxHeight uint64) (exported.ConsensusState, bool) {
-	for i := maxHeight; i > 0; i-- {
-		found := k.HasClientConsensusState(ctx, clientID, i)
+func (k Keeper) GetClientConsensusStateLTE(ctx sdk.Context, clientID string, maxHeight exported.Height) (exported.ConsensusState, bool) {
+	// NOTE: For tendermint Heights this is only implemented for a single epoch
+	height := maxHeight
+	var success bool
+	for height.Valid() {
+		found := k.HasClientConsensusState(ctx, clientID, height)
```

Comment:
Hmm, this isn't very efficient. Can we iterate on the key?

Reply:
I can look into this on a separate PR. Should be able to use `iavl.ReverseIterator`